miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultRunService.java

```diff
@@ -506,4 +527,160 @@ public void setRunDao(RunStore runDao) {
+      }
+    } else {
+      if (!run.didSomeoneElseChangeColumn("health", user) && run.getHealth() != dto.getHealthType()) {
+        // A human user has never change the health of this run, so we will.
+        run.setHealth(dto.getHealthType());
+        isMutated = true;
+      }
+    }
+
+    final Predicate<SequencingParameters> filterParameters;
+    switch (dto.getPlatformType()) {
+    case ILLUMINA:
+      isMutated |= updateField(((IlluminaNotificationDto) dto).getCallCycle(), ((IlluminaRun) run).getCallCycle(),
+          v -> ((IlluminaRun) run).setCallCycle(v));
+      isMutated |= updateField(((IlluminaNotificationDto) dto).getImgCycle(), ((IlluminaRun) run).getImgCycle(),
+          v -> ((IlluminaRun) run).setImgCycle(v));
+      isMutated |= updateField(((IlluminaNotificationDto) dto).getNumCycles(), ((IlluminaRun) run).getNumCycles(),
+          v -> ((IlluminaRun) run).setNumCycles(v));
+      isMutated |= updateField(((IlluminaNotificationDto) dto).getScoreCycle(), ((IlluminaRun) run).getScoreCycle(),
+          v -> ((IlluminaRun) run).setScoreCycle(v));
+      isMutated |= updateField(dto.isPairedEndRun(), run.getPairedEnd(), run::setPairedEnd);
+      // TODO filter the parameters based on the properties
+      filterParameters = p -> true;
+      break;
+    case IONTORRENT:
+      // Nothing to do
+      filterParameters = p -> true;
+      break;
+    case LS454:
+      isMutated |= updateField(((LS454NotificationDto) dto).getCycles(), ((LS454Run) run).getCycles(),
+          v -> ((LS454Run) run).setCycles(v));
+      isMutated |= updateField(dto.isPairedEndRun(), run.getPairedEnd(), run::setPairedEnd);
+      filterParameters = p -> true;
+      break;
+    case OXFORDNANOPORE:
+      throw new NotImplementedException();
+    case PACBIO:
+      isMutated |= updateField(((PacBioNotificationDto) dto).getMovieDurationInSec(), ((PacBioRun) run).getMovieDuration(),
+          v -> ((PacBioRun) run).setMovieDuration(v));
+      filterParameters = p -> true;
+      break;
+    case SOLID:
+      // Nothing to do
+      filterParameters = p -> true;
+      break;
+    default:
+      throw new NotImplementedException();
+    }
+
+    // If the sequencing parameters haven't been updated by a human, see if we can find exactly one that matches.
+    if (!run.didSomeoneElseChangeColumn("parameters", user)) {
+      List<SequencingParameters> possibleParameters = sequencingParametersService.getForPlatform(sequencer.getPlatform().getId()).stream()
+          .filter(parameters -> !parameters.getName().startsWith("Custom")).filter(filterParameters).collect(Collectors.toList());
+      if (possibleParameters.size() == 1) {
+        isMutated |= run.getSequencingParameters() == null || possibleParameters.get(0).getId() != run.getSequencingParameters().getId();
+        run.setSequencingParameters(possibleParameters.get(0));
+      }
+    }
+
+    if (isNew) {
+      create(run);
+    } else if (isMutated) {
+      update(run);
```

Comment:
Call `save(Run)` to skip DB fetches and other unnecessary things that happen in `update(Run)`. The call to `setChangeDetails` should be moved from `create` and `update` methods to the start of `save` to support this